### PR TITLE
Add withTokenBalances and pagination to getOwnersForContract()

### LINKED ISSUE #115: getOwnersForContract Function doesn't support pagination
As per docs This API supports pagination https://docs.alchemy.com/reference/getownersforcollection,
But SDK function getOwnersForContract doesn't support pagination.

### PR DESCRIPTION
**Pull request by thebrianchen**
Fixes #115.

This PR adds support for `withTokenBalances` and the resulting `pageKey`.

### CODE REVIEW

File: src/internal/nft-api.ts [hunk 1]
```diff
@@ -199,17 +203,45 @@ export async function* getNftsForContractIterator(
...
+  // TODO: remove `any` cast after backend is updated
+  const response: any = await requestHttpWithBackoff<
     GetOwnersForNftContractAlchemyParams,
     RawGetOwnersForContractResponse
   >(config, AlchemyApiType.NFT, 'getOwnersForCollection', srcMethod, {
+    ...options,
     contractAddress
   });
 
+  // HACK: cast `balance` field to string until NFT API makes the change
```
Comment (deric-alchemy) [inline src/internal/nft-api.ts:222]: Are we saving "TODO changes contingent on backend updates" anywhere in a list such that we can reference them (say, if you're out or end up on another project)?

**Comment by deric-alchemy** [comment]
Why did we add withTokenBalances in the first place - customer requests? Seems like an odd feature in my head

File: src/internal/nft-api.ts (hunk 1 above)
Comment (thebrianchen) [reply src/internal/nft-api.ts:222]: Ah this one isn't big enough to warrant an asana task, but should be wrapped up by @ayang-rkjs soon!

**Comment by thebrianchen** [comment]
> Why did we add withTokenBalances in the first place - customer requests? Seems like an odd feature in my head

The old endpoint only returned a giant list of wallet addresses, which made it hard to figure out who owned which token, or how many of tokens a wallet owned. Finding that information for multiple owners is a pain, so this option returns all the info at once for the end developer to prase.

**Comment by thebrianchen** [comment]
Quick update, after discussing on slack, i'm going to revert the balance to string conversion and instead leave it as number in order to match `getNftsForOwner()`.

**Review by dphilipson — APPROVED** [review]

**Pull request merged**